### ISSUE TITLE
E001: run density geometry on neurOS longitudinal EEG authority

### ISSUE THREAD

**Issue by sidhulyalkar** [sidhulyalkar/QuantumBCI#3]
## Goal

Execute the first real QuantumBCI density-geometry study on **exact neurOS longitudinal evidence authority**, beginning with Kumar2024 and expanding only after the full pipeline is qualified.

This issue is the execution follow-up to QuantumBCI PR #2. It should not weaken or duplicate neurOS's split/provenance contracts.

## Dependencies

- [ ] QuantumBCI PR #2 merged: v0.3 manifests/orchestration + neurOS bridge
- [ ] neurOS PR #28 merged or an equivalent stable sample-authority API exists
- [ ] Pin exact QuantumBCI and neurOS source SHAs for the study
- [ ] Record upstream dataset/version checksum separately from neurOS partition fingerprints

Until neurOS PR #28 is stable, use only already-merged `GroupedEvaluationData`, `chronological_partition`, `NestedCalibrationSplit`, and related public contracts.

## Phase A — frozen evidence authority

For each participant and target session:

1. collect Kumar2024 through the neurOS/MOABB evidence adapter;
2. preserve participant/session/run identities;
3. train only on chronologically prior sessions;
4. freeze target-session calibration pool and final evaluation set once;
5. serialize raw-data checksum + neurOS partition/calibration authority;
6. make every representation/method consume the same case membership.

Hard failure if any final-evaluation example influences preprocessing, representation fitting, target adaptation, SourceWeigher target summaries, hyperparameter selection, or calibration.

## Phase B — representation cache

Prepare immutable representations with per-artifact SHA-256 and provenance:

### neurOS-native controls
- CSP / Riemannian covariance baseline where applicable
- EEGNet representation
- EEG-Conformer representation
- frozen versions where the neurOS authority exposes them

### neurOS foundation lane
- runnable `neuros-foundation` registry adapters only
- fail closed for catalog-only/unavailable models

### external foundation replication lanes
- LaBraM
- EEGPT

Do not compare methods on separately recomputed nominally equivalent embeddings. If SourceWeigher and unweighted transfer are compared, they must consume the same frozen tensors.

## Phase C — representation hypotheses

On each exact embedding tensor compare:

1. pooled/linear latent control
2. PCA control
3. covariance / Riemannian tangent-space control
4. bilinear Gram control
5. random-PSD control matched in output dimension
6. QuantumBCI density vector
7. low-capacity density observables

Readout family/capacity, fitting budget, calibration examples and evaluation cases must be matched wherever mathematically possible.

## Phase D — SourceWeigher adversary

Use stable `neuros-sourceweigher.RepresentationSourceWeigher` as a reliability-aware adaptation control.

Critical semantics:

- target summaries may use **declared calibration examples only**;
- final evaluation examples never enter target distribution estimation;
- zero-calibration target-dependent SourceWeigher is unavailable rather than approximated with evaluation data;
- compare weighted vs unweighted transfer on the exact same embeddings.

Question being tested: does density geometry add value beyond simply weighting historical sessions/sources more intelligently?

## Phase E — mechanism interventions

Run through the shared `neuros-mechint` input-evidence framework:

- `RemoveDensityOffDiagonals`
- `PermuteDensityBasis`
- `MixWithMaximallyMixedState(alpha)` dose response

Also include complexity-matched control interventions that perturb an equal number of representation degrees of freedom without specifically targeting the density mechanism.

A predictive gain without mechanism-specific intervention sensitivity is not promoted as density-specific evidence.

## Statistics

Primary unit of inference: participant/case, not trial.

Report:

- balanced accuracy
- macro F1 where label geometry supports it
- calibration error / Brier score
- per-subject paired deltas
- hierarchical or participant-level bootstrap CIs
- calibration-frontier curves using fixed final evaluation examples
- representation/compute cost
- mechanism-observable stability across seeds/resamples

Promotion gate from E001 manifest remains authoritative. Do not tune the gate after inspecting results.

## Required artifact bundle

```text
study_manifest.json
source_revisions.json
dataset_fingerprint.json
neuros_authority.json
embedding_index.json
representation_index.json
predictions.parquet
fit_summary.json
sourceweigher_diagnostics.json
mechanism_observables.json
mechanism_interventions.json
bootstrap_metrics.json
evidence_ledger.json
artifact_hashes.json
report.md
```

Every artifact should be attributable to a specific QuantumBCI source SHA, neurOS source SHA, dataset checksum, frozen authority fingerprint, representation fingerprint, config and seed.

## Expansion order

Only after Kumar2024 completes cleanly:

1. Ma2020 longitudinal drift/stability replication
2. Wang2026 repeated-session/online-context replication
3. TUEV/TUAB or another external task family for out-of-ecosystem replication
4. ORION token representation lane after raw/foundation controls are understood

## Scientific interpretation ceiling

Even a successful E001 supports only the **quantum-inspired density-geometry** claim class. It does not constitute evidence that neural tissue is physically quantum.

**Comment by sidhulyalkar** [comment]
v0.4 execution foundation is now qualified in PR #4 (`71f4eeaa8dcc625940133444d8a5814f0c24f08a`, Actions run `32775454097`).

Issue #3 should build on these newly stable surfaces rather than creating a separate runner:

- `IndexSplit.from_partition(...)` for stable neurOS evidence partitions;
- `benchmark_density_embeddings(...)` for the matched density/diagonal/pooled/off-diagonal-intervention benchmark;
- `WorkbenchConfig` + `RunStore` for artifact identity and inspection;
- the packaged `E001_density_geometry` manifest, addressable by stable experiment ID;
- CLI/JSON conventions from `quantumbci benchmark`, `experiments plan`, and `runs`;
- existing v0.3 `NeurOSEvidenceBinding` for raw-data + split + source-revision scientific identity.

For the real Kumar2024 runner, preserve the stronger issue requirements: use the frozen neurOS authority, add raw-data checksums, reuse exact embedding tensors across weighted/unweighted/QuantumBCI lanes, and record SourceWeigher + neuros-mechint evidence in the same run bundle. Synthetic `quantumbci smoke` remains software/mechanism sanity only and must not be mixed into empirical promotion evidence.

**Comment by sidhulyalkar** [comment]
## E001 execution boundary updated after v0.6 + v0.6.1

The implementation assumptions in this issue have materially changed in a scientifically useful way.

### Landed boundaries

- **v0.6 / PR #6** established that the current constructor `rho = X^H X / Tr(X^H X)` is exactly a trace-normalized Hermitian second moment. The density representation is therefore **not eligible for representation-information novelty** relative to exact normalized covariance.
- **v0.6.1 / PR #7** is now merged at `d9c5daadcddbd4d169607d436fe709f94e1a6b77` and provides the package-owned Kumar2024 execution path under the merged neurOS `LongitudinalCaseAuthority`.
- Exact neurOS revision used for qualification: `ffa28ed552dc75158b673fdcd70729b1c9c69b47`.

### Kumar2024 execution contract now available

The installed command is `quantumbci-kumar2024`.

The first real lane uses MOABB epochs as genuine **time tokens × EEG-channel features** and reproduces neurOS prior-session chronology, deterministic case seeds, nested target calibration and immutable final evaluation authority.

Raw identity mirrors the current MOABB loader:

- one shared extracted Zenodo root;
- MOABB 1–9 → raw 1–9;
- MOABB 10–18 → raw 11–19;
- hash only selected Offline/Online bar-feedback GDFs;
- exclude `Race/**`;
- each participant case binds its participant-specific raw fingerprint;
- the study binds the aggregate selected-cohort fingerprint.

Representation adaptation is now frozen explicitly:

- sample-local density/covariance/operator features prepared once per participant tensor;
- flattened PCA fit once on **chronological source history only** for each target case;
- target calibration updates matched readouts only;
- final evaluation never enters representation fitting.

The finished bundle records this contract in `run.json`, `study_manifest.json`, `representation_index.json`, `evidence_ledger.json` and `report.md`, then re-closes the SHA-256 artifact ledger and supports direct RO-Crate export.

### Qualification

Exact PR #7 head `ba1f72b3f2cf1e7fb0431ed39a0ca8a14729220d` passed:

- generic run `32790307016`: Python 3.10/3.11/3.12, workbench/export/wheel and merged-neurOS integration;
- Kumar contract run `32790307021`: installed study surface, real neurOS authority classes, cached/source-history-frozen E001 execution, finalized evidence bundle, participant inference, RO-Crate export and wheel contents.

### Real empirical ladder

No real Kumar2024 result is claimed yet. The repository now contains a **manual-only** workflow named `E001 Kumar2024 real study` so normal PRs/pushes cannot download the public dataset.

Run in this order:

1. `smoke`: subjects 1 + 10, target session 5;
2. inspect raw/processed/authority/representation fingerprints, results and report;
3. `cohort`: all 18 participants, target session 5;
4. inspect GR/PAR effects and participant-level deltas;
5. `full-longitudinal`: all 18 participants, target sessions 1–5.

The density-minus-normalized-covariance result should remain exactly zero. The empirical value of E001 is now in validating the evidence machinery and characterizing normalization/operator/control behavior under longitudinal drift. A genuinely new quantum-structured claim must come from a **non-equivalent downstream operator, dynamical or contextual mechanism**, not from the current density representation itself.

Keep this issue open until the real smoke/cohort artifacts have been executed and reviewed.

**Comment by sidhulyalkar** [comment]
Repository cleanup note (v0.19): this execution issue has been superseded by the newer BMRB-Representation program in #20 and the confirmatory/known-truth infrastructure now merged through v0.19. The original goals around exact neurOS authority, frozen representation lanes, matched classical controls, intervention evidence, and participant-level inference are now represented by the newer BMRB workflow and issues. Closing this older tracker to keep the roadmap unambiguous; any still-relevant empirical execution work should continue in #20.
